src/core/App.js

```diff
@@ -13,10 +14,12 @@ import customRoutes from "./routes";
 
 import Charts from "./pages/Charts";
 import Layout from "./common/CustomLayout";
+import InitializePage from "./pages/InitializePage";
+import { themeCommonElements } from "../version/config/theme.config";
 
 const dataProvider = customDataProvider();
-
 const plugins = corePlugins.concat(nonCorePlugins);
+const Homepage = get(themeCommonElements, 'homePage', Charts);
```

Comment:
again, you should do an inversion of control here

Reply:
@fzaninotto 
Francois, could you please explain what do you mean in "inversion of control"?

I changed "lodash" to "lodash/get" everywhere

Reply:
instead of asking another function to return a component, consider that this App component will be rendered by other (specific) components. It's the responsibility of these components to provide the subcomponents they want to adapt the look and feel.

e.g.

```
// in packages/helm/App.js
import { App } from 'pulse-tile-core';
const HelmApp = () => <App homepage={HelmHomePage} appBar={HelmAppBar} />
export default HelmApp;

// in packages/NHS/App.js
import { App } from 'pulse-tile-core';
const NHSApp = () => <App homepage={NHSHomePage} appBar={NHSAppBar} />
export default NHSApp;
```

instead of 

```
// in packages/pulse-tile-core/App.js
const App = () => <App homepage={getHomePage()} appBar={getAppBar()} />
```

I call it an *inversion of control* because it's not up to the core component to decide which subcomponents to use anymore - this responsibility is shifted to its parent.